File: backend/src/main/java/it/unical/inf/ea/backend/dto/creation/OrderCreateDTO.java
package it.unical.inf.ea.backend.dto.creation;

import it.unical.inf.ea.backend.dto.AddressDTO;
import it.unical.inf.ea.backend.dto.CartDTO;
import it.unical.inf.ea.backend.dto.CartItemDTO;
import it.unical.inf.ea.backend.dto.PaymentMethodDTO;
import it.unical.inf.ea.backend.dto.basics.ProductBasicDTO;
import it.unical.inf.ea.backend.dto.enums.OrderStatus;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor
@ToString
@Builder
@AllArgsConstructor
public class OrderCreateDTO {

    @NotNull
    private List<CartItemDTO> items;

    @NotNull
    private String addressId;

    @NotNull
    private String paymentMethodId;

    @NotNull
    private String userId;

}

File: frontend/app/src/main/java/com/android/frontend/dto/creation/OrderCreateDTO.kt
package com.android.frontend.dto.creation

import com.android.frontend.dto.AddressDTO
import com.android.frontend.dto.CartDTO
import com.android.frontend.dto.CartItemDTO
import com.android.frontend.dto.PaymentMethodDTO

data class OrderCreateDTO(
    val items : List<CartItemDTO>,
    val addressId : String,
    val paymentMethod : String,
    val userId : String
)